scripts/bash/mergeEnvForDB.sh

```diff
@@ -21,30 +21,30 @@ CALLING_SCRIPT_DIR=$(get_caller_package_directory)
 CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 # Fixed paths to the .env files for the test db
-file1="$CURRENT_DIR/../../env/db.env"
-file2="$CURRENT_DIR/../../env/pg.env"
-file3="$CURRENT_DIR/../../env/data-lake.env"
-file4="$CALLING_SCRIPT_DIR/.env"
-
-common_files="$file1 $file2 $file3 $file4"
-
- # Remove files that don't exist
-for file in $common_files; do
-    if [[ ! -f $file ]]; then
-        common_files=$(echo "$common_files" | sed "s|$file||g")
-    fi
-done
-
-# Load environment variables from .env files
-merged_content="$(cat $common_files)" 
+common_files=(
+    "./../../env/db.env"
+    "./../../env/pg.env"
+    "./../../env/data-lake.env"
+    "./.env"
+)
+
+# Load environment variables from .env files (if they exist)
+merged_content=$(
+	for file in "${common_files[@]}"; do
+		if [[ -f $file ]]; then
+			cat "$file"
+		fi
+	done
+)
```

Comment:
The following _looks_ perfectly fine, but breaks because when you remove an element from a Bash array, the array doesn’t get shorter. It just leaves an “empty” element at that index. This results in `cat ''` which fails because the `''` file (obviously) doesn’t exist

```bash
# Remove files that don't exist
for file in "${common_files[@]}"; do
    if [[ ! -f $file ]]; then
        common_files=("${common_files[@]/$file}")
    fi
done

merged_content=$(cat "${common_files[@]}")
```